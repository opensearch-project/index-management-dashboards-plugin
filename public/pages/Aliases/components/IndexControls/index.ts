/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexControls from "./IndexControls";

export * from "./IndexControls";
export default IndexControls;
