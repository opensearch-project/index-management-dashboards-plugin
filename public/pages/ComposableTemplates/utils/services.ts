/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { CommonService } from "../../../services";
import { TemplateItemRemote } from "../../../../models/interfaces";

export const getAllUsedComponents = async ({ commonService }: { commonService: CommonService }) => {
  const allTemplatesResponse = await commonService.apiCaller<{
    index_templates?: Array<{
      name: string;
      index_template: TemplateItemRemote;
    }>;
  }>({
    endpoint: "transport.request",
    data: {
      method: "GET",
      path: "/_index_template/*",
    },
  });

  if (allTemplatesResponse && allTemplatesResponse.ok) {
    return allTemplatesResponse.response.index_templates || [];
  }

  return [];
};
