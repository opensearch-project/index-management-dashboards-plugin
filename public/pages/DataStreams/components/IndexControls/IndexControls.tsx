/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useEffect, useState } from "react";
import { EuiCompressedFieldSearch, EuiFlexGroup, EuiFlexItem } from "@elastic/eui";

export interface SearchControlsProps {
  value: {
    search: string;
  };
  onSearchChange: (args: SearchControlsProps["value"]) => void;
  useNewUX?: boolean;
}

export default function SearchControls(props: SearchControlsProps) {
  const [state, setState] = useState<SearchControlsProps["value"]>(props.value);
  const onChange = <T extends keyof SearchControlsProps["value"]>(field: T, value: SearchControlsProps["value"][T]) => {
    const payload = {
      ...state,
      [field]: value,
    };
    setState(payload);
    props.onSearchChange(payload);
  };
  useEffect(() => {
    setState(props.value);
  }, [props.value]);
  return (
    <EuiFlexGroup gutterSize="s" alignItems="center">
      <EuiFlexItem>
        <EuiCompressedFieldSearch
          fullWidth
          placeholder="Search..."
          value={state.search}
          onChange={(e) => onChange("search", e.target.value)}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}
