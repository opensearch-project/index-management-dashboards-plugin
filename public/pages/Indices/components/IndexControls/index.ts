/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import IndexControls from "./IndexControls";

export default IndexControls;
