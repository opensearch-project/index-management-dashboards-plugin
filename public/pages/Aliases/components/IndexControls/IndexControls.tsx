/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useEffect, useState } from "react";
import { EuiComboBox, EuiFieldSearch, EuiFlexGroup, EuiFlexItem } from "@elastic/eui";
import { ALIAS_STATUS_OPTIONS } from "../../../../utils/constants";

export interface SearchControlsProps {
  value: {
    search: string;
    status: string;
  };
  onSearchChange: (args: SearchControlsProps["value"]) => void;
}

export default function SearchControls(props: SearchControlsProps) {
  const [state, setState] = useState<SearchControlsProps["value"]>(props.value);
  const onChange = <T extends keyof SearchControlsProps["value"]>(field: T, value: SearchControlsProps["value"][T]) => {
    const payload = {
      ...state,
      [field]: value,
    };
    setState(payload);
    props.onSearchChange(payload);
  };
  useEffect(() => {
    setState(props.value);
  }, [props.value]);
  return (
    <EuiFlexGroup style={{ padding: "0px 5px" }} alignItems="center">
      <EuiFlexItem>
        <EuiFieldSearch fullWidth placeholder="Search..." value={state.search} onChange={(e) => onChange("search", e.target.value)} />
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <EuiComboBox
          style={{
            width: 300,
          }}
          placeholder="Status"
          options={ALIAS_STATUS_OPTIONS}
          selectedOptions={state.status ? state.status.split(",").map((label) => ({ label })) : []}
          onChange={(val) => onChange("status", (val || []).map((item) => item.label).join(","))}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}
