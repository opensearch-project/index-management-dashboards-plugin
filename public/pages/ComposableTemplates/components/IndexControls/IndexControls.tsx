/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useEffect, useState } from "react";
import { EuiButton, EuiFieldSearch, EuiFlexGroup, EuiFlexItem } from "@elastic/eui";
import FilterGroup from "../../../../components/FilterGroup";
import { IndicesUpdateMode } from "../../../../utils/constants";

export interface SearchControlsProps {
  value: {
    search: string;
    selectedTypes: string[];
  };
  useNewUX?: boolean;
  onSearchChange: (args: SearchControlsProps["value"]) => void;
}

export default function SearchControls(props: SearchControlsProps) {
  const [state, setState] = useState<SearchControlsProps["value"]>(props.value);
  const onChange = <T extends keyof SearchControlsProps["value"]>(field: T, value: SearchControlsProps["value"][T]) => {
    const payload = {
      ...state,
      [field]: value,
    };
    setState(payload);
    props.onSearchChange(payload);
  };
  useEffect(() => {
    setState(props.value);
  }, [props.value]);
  return (
    <EuiFlexGroup style={{ padding: "0px 5px" }} alignItems="center">
      <EuiFlexItem>
        <EuiFieldSearch
          compressed={props.useNewUX}
          fullWidth
          placeholder="Search..."
          value={state.search}
          onChange={(e) => onChange("search", e.target.value)}
        />
      </EuiFlexItem>
      <EuiFlexItem style={{ flexBasis: "100px" }} grow={false}>
        <FilterGroup
          filterButtonProps={{
            children: "Types",
          }}
          onChange={(val) => onChange("selectedTypes", val || [])}
          value={state.selectedTypes}
          useNewUx={props.useNewUX}
          options={[
            {
              label: IndicesUpdateMode.alias,
            },
            {
              label: IndicesUpdateMode.mappings,
            },
            {
              label: IndicesUpdateMode.settings,
            },
          ]}
        />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}
