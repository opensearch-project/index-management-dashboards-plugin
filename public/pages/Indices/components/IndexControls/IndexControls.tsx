/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { Component } from "react";
import {
  ArgsWithError,
  ArgsWithQuery,
  EuiButton,
  EuiButtonIcon,
  EuiFlexGroup,
  EuiSpacer,
  EuiFlexItem,
  EuiSearchBar,
  EuiSwitch,
  EuiButtonEmpty,
} from "@elastic/eui";
import { DataStream, ManagedCatIndex } from "../../../../../server/models/interfaces";
import IndicesActions from "../../containers/IndicesActions";
import { getUISettings } from "../../../../services/Services";

interface IndexControlsProps {
  search: string;
  showDataStreams: boolean;
  onSearchChange: (args: ArgsWithQuery | ArgsWithError) => void;
  onRefresh: () => Promise<void>;
  getDataStreams: () => Promise<DataStream[]>;
  toggleShowDataStreams: () => void;
  selectedItems: ManagedCatIndex[];
}

interface IndexControlsState {
  refreshInterval: number;
  isPaused: boolean;
}

export default class IndexControls extends Component<IndexControlsProps, IndexControlsState> {
  state: IndexControlsState = {
    refreshInterval: 0,
    isPaused: true,
  };

  onRefreshChange = ({ refreshInterval, isPaused }: IndexControlsState): void => {
    this.setState({ isPaused, refreshInterval });
  };

  getDataStreams = async () => {
    return (await this.props.getDataStreams()).map((ds) => ({ value: ds.name }));
  };

  render() {
    const { search, onSearchChange, showDataStreams, toggleShowDataStreams, onRefresh, selectedItems } = this.props;

    const schema = {
      strict: true,
      fields: {
        indices: {
          type: "string",
        },
        data_streams: {
          type: "string",
        },
      },
    };

    const filters = showDataStreams
      ? [
          {
            type: "field_value_selection",
            field: "data_streams",
            name: "Data streams",
            noOptionsMessage: "No data streams found",
            multiSelect: "or",
            cache: 60000,
            options: () => this.getDataStreams(),
          },
        ]
      : undefined;

    const uiSettings = getUISettings();
    const useUpdatedUX = uiSettings.get("home:useNewHomePage");

    return useUpdatedUX ? (
      <EuiFlexGroup style={{ padding: "0px 5px 16px 5px" }} alignItems="center">
        <EuiFlexItem>
          <EuiSearchBar
            compressed
            query={search}
            box={{ placeholder: "Search", schema, incremental: true, compressed: true }}
            onChange={onSearchChange}
            filters={filters}
          />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiButtonIcon iconType="refresh" data-test-subj="refreshButton" display="base" size="s" />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <IndicesActions
            {...this.props}
            onDelete={onRefresh}
            onClose={onRefresh}
            onShrink={onRefresh}
            selectedItems={selectedItems}
            getIndices={onRefresh}
          />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiSwitch
            compressed
            label="Show data stream indexes"
            checked={showDataStreams}
            onChange={toggleShowDataStreams}
            data-test-subj="toggleShowDataStreams"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    ) : (
      <EuiFlexGroup style={{ padding: "0px 5px" }} alignItems="center">
        <EuiFlexItem>
          <EuiSearchBar
            query={search}
            box={{ placeholder: "Search", schema, incremental: true }}
            onChange={onSearchChange}
            filters={filters}
          />
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiSwitch
            label="Show data stream indexes"
            checked={showDataStreams}
            onChange={toggleShowDataStreams}
            data-test-subj="toggleShowDataStreams"
          />
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  }
}
