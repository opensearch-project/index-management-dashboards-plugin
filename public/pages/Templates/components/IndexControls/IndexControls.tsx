/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useEffect, useState } from "react";
import { EuiFieldSearch, EuiFlexGroup, EuiFlexItem } from "@elastic/eui";
import { getUISettings } from "../../../../services/Services";
import TemplatesActions from "../../containers/TemplatesActions";
import { ITemplate } from "../../interface";
import { RouteComponentProps } from "react-router-dom";

export interface SearchControlsProps {
  value: {
    search: string;
  };
  onSearchChange: (args: SearchControlsProps["value"]) => void;
  selectedItems: ITemplate[];
  getTemplates: () => Promise<void>;
  history: RouteComponentProps["history"];
}

export default function SearchControls(props: SearchControlsProps) {
  const [state, setState] = useState<SearchControlsProps["value"]>(props.value);
  const onChange = <T extends keyof SearchControlsProps["value"]>(field: T, value: SearchControlsProps["value"][T]) => {
    const payload = {
      ...state,
      [field]: value,
    };
    setState(payload);
    props.onSearchChange(payload);
  };
  useEffect(() => {
    setState(props.value);
  }, [props.value]);

  const uiSettings = getUISettings();
  const useUpdatedUX = uiSettings.get("home:useNewHomePage");

  return useUpdatedUX ? (
    <EuiFlexGroup style={{ padding: "0px 5px 16px 5px" }} alignItems="center">
      <EuiFlexItem>
        <EuiFieldSearch
          compressed
          fullWidth
          placeholder="Search"
          value={state.search}
          onChange={(e) => onChange("search", e.target.value)}
        />
      </EuiFlexItem>
      <EuiFlexItem grow={false}>
        <TemplatesActions selectedItems={props.selectedItems} onDelete={props.getTemplates} history={props.history} />
      </EuiFlexItem>
    </EuiFlexGroup>
  ) : (
    <EuiFlexGroup style={{ padding: "0px 5px" }} alignItems="center">
      <EuiFlexItem>
        <EuiFieldSearch fullWidth placeholder="Search..." value={state.search} onChange={(e) => onChange("search", e.target.value)} />
      </EuiFlexItem>
    </EuiFlexGroup>
  );
}
