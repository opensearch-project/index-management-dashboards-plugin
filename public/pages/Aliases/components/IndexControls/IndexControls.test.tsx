/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, waitFor } from "@testing-library/react";
// @ts-ignore
import userEvent from "@testing-library/user-event";
import IndexControls from "./IndexControls";
import { getApplication, getNavigationUI, getUISettings } from "../../../../services/Services";

jest.mock("../../../../services/Services", () => ({
  ...jest.requireActual("../../../../services/Services"),
  getUISettings: jest.fn(),
  getApplication: jest.fn(),
  getNavigationUI: jest.fn(),
}));

beforeEach(() => {
  (getUISettings as jest.Mock).mockReturnValue({
    get: jest.fn().mockReturnValue(false), // or false, depending on your test case
  });
  (getApplication as jest.Mock).mockReturnValue({});

  (getNavigationUI as jest.Mock).mockReturnValue({});
});

describe("<IndexControls /> spec", () => {
  it("renders the component", async () => {
    const { container } = render(<IndexControls value={{ search: "testing", status: "1" }} onSearchChange={() => {}} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("onChange with right data", async () => {
    const onSearchChangeMock = jest.fn();
    const { getByTestId, getByPlaceholderText } = render(
      <IndexControls value={{ search: "", status: "" }} onSearchChange={onSearchChangeMock} />
    );

    userEvent.type(getByTestId("comboBoxSearchInput"), "closed{enter}");
    expect(onSearchChangeMock).toBeCalledTimes(1);
    expect(onSearchChangeMock).toBeCalledWith({
      search: "",
      status: "closed",
    });
    userEvent.type(getByPlaceholderText("Search..."), "test");
    await waitFor(() => {
      expect(onSearchChangeMock).toBeCalledTimes(5);
      expect(onSearchChangeMock).toBeCalledWith({
        search: "test",
        status: "closed",
      });
    });
  });
});
