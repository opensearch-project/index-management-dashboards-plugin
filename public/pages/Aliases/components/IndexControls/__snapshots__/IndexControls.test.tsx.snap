// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IndexControls /> spec renders the component 1`] = `
<div
  class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--directionRow euiFlexGroup--responsive"
  style="padding: 0px 5px;"
>
  <div
    class="euiFlexItem"
  >
    <div
      class="euiFormControlLayout euiFormControlLayout--fullWidth"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <input
          class="euiFieldSearch euiFieldSearch--fullWidth euiFieldSearch-isClearable"
          placeholder="Search..."
          type="search"
          value="testing"
        />
        <div
          class="euiFormControlLayoutIcons"
        >
          <span
            class="euiFormControlLayoutCustomIcon"
          >
            EuiIconMock
          </span>
        </div>
        <div
          class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
        >
          <button
            aria-label="Clear input"
            class="euiFormControlLayoutClearButton"
            type="button"
          >
            EuiIconMock
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiFlexItem euiFlexItem--flexGrowZero"
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      class="euiComboBox"
      role="combobox"
      style="width: 300px;"
    >
      <div
        class="euiFormControlLayout"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <div
            class="euiComboBox__inputWrap euiComboBox__inputWrap-isClearable"
            data-test-subj="comboBoxInput"
            tabindex="-1"
          >
            <span
              class="euiBadge euiBadge--hollow euiBadge--iconRight euiComboBoxPill"
              title="1"
            >
              <span
                class="euiBadge__content"
              >
                <span
                  class="euiBadge__text"
                >
                  1
                </span>
                <button
                  aria-label="Remove 1 from selection in this group"
                  class="euiBadge__iconButton"
                  title="Remove 1 from selection in this group"
                  type="button"
                >
                  EuiIconMock
                </button>
              </span>
            </span>
            <div
              class="euiComboBox__input"
              style="font-size: 14px; display: inline-block;"
            >
              <input
                aria-controls=""
                data-test-subj="comboBoxSearchInput"
                role="textbox"
                style="box-sizing: content-box; width: 2px;"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
              />
            </div>
          </div>
          <div
            class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
          >
            <button
              aria-label="Clear input"
              class="euiFormControlLayoutClearButton"
              data-test-subj="comboBoxClearButton"
              type="button"
            >
              EuiIconMock
            </button>
            <button
              aria-label="Open list of options"
              class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
              data-test-subj="comboBoxToggleListButton"
              type="button"
            >
              EuiIconMock
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
